src/background.js

```diff
@@ -57,13 +67,21 @@ browser.messages.onUpdated.addListener(window.scrNoti.messageOnUpdatedListener);
 //==========================================
 window.scrNoti.messageDeletedListener = async (messagesObj) => {
   for (const message of messagesObj.messages) {
+    if (message.junk) {
+      continue;
+    }
+
     if (!(await window.scrNoti.isFolderToCheck(message.folder))) {
       continue;
     }
 
-    if (!message.junk && !message.read) {
-      await window.scrNoti.runUnreadValidation();
-      return;
+    seenMessages[message.folder].delete(message.id);
+    if (!message.read) {
+// FIXME: Do we ever reach this branch, because the message flag "read" is cleared and as such
```

Comment:
I think you are right. I even think we may not need the `onDeleted` listener at all, since the `onUpdated` listener is also called, as you say. 

In fact, it seems the "onDeleted" even is only called when a message is _permanently_ deleted (so deleted from the trash folder): https://webextension-api.thunderbird.net/en/91/messages.html#ondeleted . Not useful for us, I think.

Reply:
I just removed the `onDeleted` listener.